File: TOMIMSV4/Entity/Transacciones/Despacho/clsBeVW_Despacho_Rep.vb
Public Class clsBeVW_Despacho_Rep
    Implements ICloneable

    Public Property IdPickingUbic() As Integer = 0

    Public Property IdStock() As Integer = 0

    Public Property IdPedidoDet() As Integer = 0

    Public Property IdPropietarioBodega() As Integer = 0

    Public Property IdProductoBodega() As Integer = 0

    Public Property IdProductoEstado() As Integer = 0

    Public Property IdPresentacion() As Integer = 0

    Public Property IdUnidadMedida() As Integer = 0

    Public Property IdRecepcion() As Integer = 0

    Public Property IdDespachoEnc() As Integer = 0

    Public Property IdDespachoDet() As Integer = 0

    Public Property IdPedidoEnc() As Integer = 0

    Public Property Propietario() As String = ""

    Public Property Codigo_Producto() As String = ""

    Public Property Nombre_Producto() As String = ""

    Public Property UM() As String = ""

    Public Property Presentacion() As String = ""

    Public Property Factor() As Double = 0.0

    Public Property Estado() As String = ""

    Public Property Lote() As String = ""

    Public Property Licencia() As String = ""

    Public Property Vence() As Date = Date.Now

    Public Property Ubicacion_Origen() As String = ""

    Public Property Cantidad_pickeada() As Double = 0.0

    Public Property Cantidad_verificada() As Double = 0.0

    Public Property Peso_Pickeado() As Double = 0.0

    Public Property Peso_Verificado() As Double = 0.0

    Public Property CantidadDespachada() As Double = 0.0

    Public Property PesoDespachado() As Double = 0.0

    Public Property Encontrado() As Boolean = False

    Public Property Acepto() As Boolean = False

    Public Property No_Documento_WMS() As Integer = 0

    Public Property No_Referencia() As String = ""

    Public Property Codigo_Cliente() As String = ""

    Public Property Nombre_Cliente() As String = ""

    Public Property Idubicacionvirtual() As Integer = 0

    Public Property Es_bodega_recepcion() As Boolean = False

    Public Property Es_bodega_traslado() As Boolean = False

    Public Property No_pase() As Integer = 0

    Public Property Observacion() As String = ""

    Public Property Numero() As Integer = 0

    Public Property Marchamo() As String = ""

    Public Property Codigo_Ruta() As String = ""

    Public Property Nombre_Ruta() As String = ""

    Public Property Placa_Vehiculo() As String = ""

    Public Property Placa_Comercial() As String = ""

    Public Property Marca_Vehiculo() As String = ""

    Public Property Modelo_Vehiculo() As String = ""

    Public Property Nombre_Piloto() As String = ""

    Public Property Apellido_Piloto() As String = ""

    Public Property No_Carnet_Piloto() As String = ""

    Public Property No_Licencia_Piloto() As String = ""

    Public Property Fecha() As Date

    Public Property codigo_poliza_pedido() As String = ""

    '#GT23112022_1400: campos DyD
    Public Property clasificacion() As String = ""
    Public Property marca() As String = ""
    Public Property familia() As String = ""
    Public Property parametro_a() As String = ""
    Public Property parametro_b() As String = ""

    '#16082023: campos pedido-poliza
    Public Property numero_orden_pedido() As String = ""

    '#GT18082023: campos ingreso-poliza
    Public Property numero_orden_ingreso() As String = ""
    Public Property codigo_poliza_ingreso() As String = ""

    '#GT30092024: campos pedido CLC Fiscal
    Public Property codigo_regimen_salida As String = ""
    Public Property placa_contenedor_salida As String = ""
    Public Property Dua_salida As String = ""
    Public Property Talla As String = ""
    Public Property Color As String = ""

    Sub New()
    End Sub

    Sub New(ByRef IdPickingUbic As Integer, ByVal IdStock As Integer, ByVal IdPedidoDet As Integer,
                        ByVal IdPropietarioBodega As Integer, ByVal IdProductoBodega As Integer,
                        ByVal IdProductoEstado As Integer, ByVal IdPresentacion As Integer,
                        ByVal IdUnidadMedida As Integer, ByVal IdRecepcion As Integer,
                        ByVal IdDespachoEnc As Integer, ByVal IdDespachoDet As Integer,
                        ByVal IdPedidoEnc As Integer, ByVal Propietario As String,
                        ByVal Codigo_Producto As String, ByVal Nombre_Producto As String,
                        ByVal UM As String, ByVal Presentacion As String, ByVal factor As Double,
                        ByVal Estado As String, ByVal lote As String, ByVal Vence As Date,
                        ByVal Ubicacion_Origen As String, ByVal cantidad_pickeada As Double,
                        ByVal cantidad_verificada As Double, ByVal Peso_Pickeado As Double,
                        ByVal Peso_Verificado As Double, ByVal CantidadDespachada As Double,
                        ByVal PesoDespachado As Double, ByVal Encontrado As Boolean,
                        ByVal Acepto As Boolean, ByVal No_Documento_WMS As Integer,
                        ByVal No_Referencia As String, ByVal Codigo_Cliente As String,
                        ByVal Nombre_Cliente As String, ByVal idubicacionvirtual As Integer,
                        ByVal es_bodega_recepcion As Boolean, ByVal es_bodega_traslado As Boolean,
                        ByVal no_pase As Integer, ByVal observacion As String, ByVal numero As Integer,
                        ByVal marchamo As String, ByVal Codigo_Ruta As String, ByVal Nombre_Ruta As String,
                        ByVal Placa_Vehiculo As String, ByVal Placa_Comercial As String,
                        ByVal Marca_Vehiculo As String, ByVal Modelo_Vehiculo As String,
                        ByVal Nombre_Piloto As String, ByVal Apellido_Piloto As String,
                        ByVal No_Carnet_Piloto As String, ByVal No_Licencia_Piloto As String,
                        ByVal Fecha As Date, ByVal codigo_poliza_pedido As String,
                        ByVal numero_orden_pedido As String,
                        ByVal codigo_poliza_ingreso As String,
                        ByVal numero_orden_ingreso As String)

        Me.IdPickingUbic = IdPickingUbic
        Me.IdStock = IdStock
        Me.IdPedidoDet = IdPedidoDet
        Me.IdPropietarioBodega = IdPropietarioBodega
        Me.IdProductoBodega = IdProductoBodega
        Me.IdProductoEstado = IdProductoEstado
        Me.IdPresentacion = IdPresentacion
        Me.IdUnidadMedida = IdUnidadMedida
        Me.IdRecepcion = IdRecepcion
        Me.IdDespachoEnc = IdDespachoEnc
        Me.IdDespachoDet = IdDespachoDet
        Me.IdPedidoEnc = IdPedidoEnc
        Me.Propietario = Propietario
        Me.Codigo_Producto = Codigo_Producto
        Me.Nombre_Producto = Nombre_Producto
        Me.UM = UM
        Me.Presentacion = Presentacion
        Me.Factor = factor
        Me.Estado = Estado
        Me.Lote = lote
        Me.Vence = Vence
        Me.Ubicacion_Origen = Ubicacion_Origen
        Me.Cantidad_pickeada = cantidad_pickeada
        Me.Cantidad_verificada = cantidad_verificada
        Me.Peso_Pickeado = Peso_Pickeado
        Me.Peso_Verificado = Peso_Verificado
        Me.CantidadDespachada = CantidadDespachada
        Me.PesoDespachado = PesoDespachado
        Me.Encontrado = Encontrado
        Me.Acepto = Acepto
        Me.No_Documento_WMS = No_Documento_WMS
        Me.No_Referencia = No_Referencia
        Me.Codigo_Cliente = Codigo_Cliente
        Me.Nombre_Cliente = Nombre_Cliente
        Me.Idubicacionvirtual = idubicacionvirtual
        Me.Es_bodega_recepcion = es_bodega_recepcion
        Me.Es_bodega_traslado = es_bodega_traslado
        Me.No_pase = no_pase
        Me.Observacion = observacion
        Me.Numero = numero
        Me.Marchamo = marchamo
        Me.Codigo_Ruta = Codigo_Ruta
        Me.Nombre_Ruta = Nombre_Ruta
        Me.Placa_Vehiculo = Placa_Vehiculo
        Me.Placa_Comercial = Placa_Comercial
        Me.Marca_Vehiculo = Marca_Vehiculo
        Me.Modelo_Vehiculo = Modelo_Vehiculo
        Me.Nombre_Piloto = Nombre_Piloto
        Me.Apellido_Piloto = Apellido_Piloto
        Me.No_Carnet_Piloto = No_Carnet_Piloto
        Me.No_Licencia_Piloto = No_Licencia_Piloto
        Me.Fecha = Fecha
        Me.codigo_poliza_pedido = codigo_poliza_pedido
        Me.numero_orden_pedido = numero_orden_pedido
        Me.codigo_poliza_ingreso = codigo_poliza_ingreso
        Me.numero_orden_ingreso = numero_orden_ingreso
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/VW_Despacho_Rep/clsBeVW_Despacho_Rep.cs
namespace WMS.EntityCore.VW_Despacho_Rep
{
    public class clsBeVW_Despacho_Rep : ICloneable
    {
        public int IdPickingUbic { get; set; } = 0;
        public int IdStock { get; set; } = 0;
        public int IdPedidoDet { get; set; } = 0;
        public int IdPropietarioBodega { get; set; } = 0;
        public int IdProductoBodega { get; set; } = 0;
        public int IdProductoEstado { get; set; } = 0;
        public int IdPresentacion { get; set; } = 0;
        public int IdUnidadMedida { get; set; } = 0;
        public int IdRecepcion { get; set; } = 0;
        public int IdDespachoEnc { get; set; } = 0;
        public int IdDespachoDet { get; set; } = 0;
        public int IdPedidoEnc { get; set; } = 0;
        public string Propietario { get; set; } = "";
        public string Codigo_Producto { get; set; } = "";
        public string Nombre_Producto { get; set; } = "";
        public string UM { get; set; } = "";
        public string Presentacion { get; set; } = "";
        public double Factor { get; set; } = 0.0;
        public string Estado { get; set; } = "";
        public string Lote { get; set; } = "";
        public string Licencia { get; set; } = "";
        public DateTime Vence { get; set; } = DateTime.Now;
        public string Ubicacion_Origen { get; set; } = "";
        public double Cantidad_pickeada { get; set; } = 0.0;
        public double Cantidad_verificada { get; set; } = 0.0;
        public double Peso_Pickeado { get; set; } = 0.0;
        public double Peso_Verificado { get; set; } = 0.0;
        public double CantidadDespachada { get; set; } = 0.0;
        public double PesoDespachado { get; set; } = 0.0;
        public bool Encontrado { get; set; } = false;
        public bool Acepto { get; set; } = false;
        public int No_Documento_WMS { get; set; } = 0;
        public string No_Referencia { get; set; } = "";
        public string Codigo_Cliente { get; set; } = "";
        public string Nombre_Cliente { get; set; } = "";
        public int Idubicacionvirtual { get; set; } = 0;
        public bool Es_bodega_recepcion { get; set; } = false;
        public bool Es_bodega_traslado { get; set; } = false;
        public int No_pase { get; set; } = 0;
        public string Observacion { get; set; } = "";
        public int Numero { get; set; } = 0;
        public string Marchamo { get; set; } = "";
        public string Codigo_Ruta { get; set; } = "";
        public string Nombre_Ruta { get; set; } = "";
        public string Placa_Vehiculo { get; set; } = "";
        public string Placa_Comercial { get; set; } = "";
        public string Marca_Vehiculo { get; set; } = "";
        public string Modelo_Vehiculo { get; set; } = "";
        public string Nombre_Piloto { get; set; } = "";
        public string Apellido_Piloto { get; set; } = "";
        public string No_Carnet_Piloto { get; set; } = "";
        public string No_Licencia_Piloto { get; set; } = "";
        public DateTime Fecha { get; set; }
        public string codigo_poliza_pedido { get; set; } = "";
        public string clasificacion { get; set; } = "";
        public string marca { get; set; } = "";
        public string familia { get; set; } = "";
        public string parametro_a { get; set; } = "";
        public string parametro_b { get; set; } = "";
        public string numero_orden_pedido { get; set; } = "";
        public string numero_orden_ingreso { get; set; } = "";
        public string codigo_poliza_ingreso { get; set; } = "";
        public string codigo_regimen_salida { get; set; } = "";
        public string placa_contenedor_salida { get; set; } = "";
        public string Dua_salida { get; set; } = "";
        public string Talla { get; set; } = "";
        public string Color { get; set; } = "";

        public clsBeVW_Despacho_Rep()
        {
        }

        public clsBeVW_Despacho_Rep(ref int IdPickingUbic, int IdStock, int IdPedidoDet,
                            int IdPropietarioBodega, int IdProductoBodega,
                            int IdProductoEstado, int IdPresentacion,
                            int IdUnidadMedida, int IdRecepcion,
                            int IdDespachoEnc, int IdDespachoDet,
                            int IdPedidoEnc, string Propietario,
                            string Codigo_Producto, string Nombre_Producto,
                            string UM, string Presentacion, double factor,
                            string Estado, string lote, DateTime Vence,
                            string Ubicacion_Origen, double cantidad_pickeada,
                            double cantidad_verificada, double Peso_Pickeado,
                            double Peso_Verificado, double CantidadDespachada,
                            double PesoDespachado, bool Encontrado,
                            bool Acepto, int No_Documento_WMS,
                            string No_Referencia, string Codigo_Cliente,
                            string Nombre_Cliente, int idubicacionvirtual,
                            bool es_bodega_recepcion, bool es_bodega_traslado,
                            int no_pase, string observacion, int numero,
                            string marchamo, string Codigo_Ruta, string Nombre_Ruta,
                            string Placa_Vehiculo, string Placa_Comercial,
                            string Marca_Vehiculo, string Modelo_Vehiculo,
                            string Nombre_Piloto, string Apellido_Piloto,
                            string No_Carnet_Piloto, string No_Licencia_Piloto,
                            DateTime Fecha, string codigo_poliza_pedido,
                            string numero_orden_pedido,
                            string codigo_poliza_ingreso,
                            string numero_orden_ingreso)
        {
            this.IdPickingUbic = IdPickingUbic;
            this.IdStock = IdStock;
            this.IdPedidoDet = IdPedidoDet;
            this.IdPropietarioBodega = IdPropietarioBodega;
            this.IdProductoBodega = IdProductoBodega;
            this.IdProductoEstado = IdProductoEstado;
            this.IdPresentacion = IdPresentacion;
            this.IdUnidadMedida = IdUnidadMedida;
            this.IdRecepcion = IdRecepcion;
            this.IdDespachoEnc = IdDespachoEnc;
            this.IdDespachoDet = IdDespachoDet;
            this.IdPedidoEnc = IdPedidoEnc;
            this.Propietario = Propietario;
            this.Codigo_Producto = Codigo_Producto;
            this.Nombre_Producto = Nombre_Producto;
            this.UM = UM;
            this.Presentacion = Presentacion;
            this.Factor = factor;
            this.Estado = Estado;
            this.Lote = lote;
            this.Vence = Vence;
            this.Ubicacion_Origen = Ubicacion_Origen;
            this.Cantidad_pickeada = cantidad_pickeada;
            this.Cantidad_verificada = cantidad_verificada;
            this.Peso_Pickeado = Peso_Pickeado;
            this.Peso_Verificado = Peso_Verificado;
            this.CantidadDespachada = CantidadDespachada;
            this.PesoDespachado = PesoDespachado;
            this.Encontrado = Encontrado;
            this.Acepto = Acepto;
            this.No_Documento_WMS = No_Documento_WMS;
            this.No_Referencia = No_Referencia;
            this.Codigo_Cliente = Codigo_Cliente;
            this.Nombre_Cliente = Nombre_Cliente;
            this.Idubicacionvirtual = idubicacionvirtual;
            this.Es_bodega_recepcion = es_bodega_recepcion;
            this.Es_bodega_traslado = es_bodega_traslado;
            this.No_pase = no_pase;
            this.Observacion = observacion;
            this.Numero = numero;
            this.Marchamo = marchamo;
            this.Codigo_Ruta = Codigo_Ruta;
            this.Nombre_Ruta = Nombre_Ruta;
            this.Placa_Vehiculo = Placa_Vehiculo;
            this.Placa_Comercial = Placa_Comercial;
            this.Marca_Vehiculo = Marca_Vehiculo;
            this.Modelo_Vehiculo = Modelo_Vehiculo;
            this.Nombre_Piloto = Nombre_Piloto;
            this.Apellido_Piloto = Apellido_Piloto;
            this.No_Carnet_Piloto = No_Carnet_Piloto;
            this.No_Licencia_Piloto = No_Licencia_Piloto;
            this.Fecha = Fecha;
            this.codigo_poliza_pedido = codigo_poliza_pedido;
            this.numero_orden_pedido = numero_orden_pedido;
            this.codigo_poliza_ingreso = codigo_poliza_ingreso;
            this.numero_orden_ingreso = numero_orden_ingreso;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
